homeassistant/components/tuya/__init__.py

```diff
@@ -105,13 +188,26 @@ async def async_setup_entry(hass, entry):
         )
         return False
 
-    hass.data[DOMAIN] = {
-        TUYA_DATA: tuya,
-        TUYA_TRACKER: None,
-        ENTRY_IS_SETUP: set(),
-        "entities": {},
-        "pending": {},
-    }
+    hass.data.setdefault(DOMAIN, {}).update(
+        {
+            TUYA_DATA: tuya,
+            TUYA_TRACKER: None,
+            ENTRY_IS_SETUP: set(),
+            "entities": {},
+            "pending": {},
+            entry.entry_id: {
+                "listener": [entry.add_update_listener(update_listener)],
+            },
+        }
+    )
+
+    _update_discovery_interval(
```

Comment:
Since this is rate limited on the tuya side, I'm not sure it makes sense to give the user control of the interval since it will most likely be ineffective below the defaults.

Reply:
Not sure that Tuya will not change this limit again in the future, hoping in lower values. In any case if the limit is set to low, a warning message is write in the log, so this option allow a tuning of the integration without requiring to modify the code.

Reply:
Generally we don't add options to change these unless there is an exceptional case.  Unless there is a history of the provider changing this frequently, we should set this to the fastest allowed polling interval allowed that doesn't flood the state machine with updates.

Reply:
Current allowed interval is very high (600 sec.). They change it within a month from 300 to 600. Lower limit that can be set now by integration option is 10 seconds, I can set to higher value that make sense for you. I think that leaving the option to reduce current interval inside a limited range should be better...

Reply:
I increased the lower limit for this options to 30 seconds